enterprise/api/v1/views.py

```diff
@@ -811,42 +811,40 @@ class PendingEnterpriseCustomerUserViewSet(EnterpriseReadWriteModelViewSet):
     """
     API views for the ``pending-enterprise-learner`` API endpoint.
     """
-
     queryset = models.PendingEnterpriseCustomerUser.objects.all()
     filter_backends = (filters.OrderingFilter, DjangoFilterBackend)
     serializer_class = serializers.PendingEnterpriseCustomerUserSerializer
     permission_classes = (permissions.IsAuthenticated, permissions.IsAdminUser)
 
     FIELDS = (
-        'enterprise_customer', 'user_email',
-    )
+          'enterprise_customer', 'user_email',
+      )
     filterset_fields = FIELDS
     ordering_fields = FIELDS
 
-    def create(self, request, *args, **kwargs):
-        serializer = self.get_serializer(data=request.data)
-        try:
-            serializer.is_valid(raise_exception=True)
-        except ValidationError as validation_error:
-            return_status = None
-            if 'user_email' in serializer.errors:
-                for error in serializer.errors['user_email']:
-                    # This error indicates that a PendingEnterpriseCustomerUser already exists.
-                    if error.code == 'unique':
-                        return_status = status.HTTP_204_NO_CONTENT
-                        break
-            elif 'non_field_errors' in serializer.errors:
-                for error in serializer.errors['non_field_errors']:
-                    # This error indicates that an EnterpriseCustomerUser already exists.
-                    if str(error) == 'EnterpriseCustomerUser record already exists':
-                        return_status = status.HTTP_204_NO_CONTENT
-                        break
-
-            if not return_status:
-                raise validation_error
-        else:
-            created = serializer.save()
+    UNIQUE = 'unique'
+    USER_EXISTS_ERROR = 'EnterpriseCustomerUser record already exists'
+
+    def _get_return_status(self, serializer, many):
```

Comment:
opinion: This is a nice pattern. I'm not thrilled with the name since it actually does the saving as well, but I would be pleased to see the handle one-or-many DRF apis in more places.

Reply:
yeaaaah, it started out as one thing and ended up as another, I'm not thrilled with the name either. But yeah, I stole the serializer `many` pattern and it seems to be OK?